lib/spack/spack/test/cmd/spec.py

```diff
@@ -50,6 +52,25 @@ def test_spec_json():
     assert 'mpich' in mpileaks
 
 
+def test_spec_format(database, config):
+    output = spec('--format', '{name}-{^mpi.name}', 'mpileaks')
+    assert output.rstrip('\n') == "mpileaks-mpich"
+
+    output = spec('--format', '{name}-{version}-{compiler.name}-{^mpi.name}',
+                  'mpileaks')
+    assert "installed package" not in output
+    assert output.rstrip('\n') == "mpileaks-2.3-gcc-mpich"
+
+    output = spec('--format', '{name}-{^mpi.name}-{hash:7}',
+                  'mpileaks')
+    output = output.rstrip('\n')
+    assert output[:-7] == "mpileaks-mpich-"
+
+    # hashes are in base32
+    for c in output[-7:]:
+        assert c in base32_alphabet
```

Comment:
IMO. You can get rid of this since I'm sure we have tests of the format feature elsewhere.